metricflow/test/model/parsing/test_metadata_parsing.py

```diff
@@ -39,6 +42,25 @@ def test_metric_metadata_parsing(simple_user_configured_model: UserConfiguredMod
         _assert_metadata_filename_is_valid(metric.metadata)
 
 
+@pytest.mark.skip("TODO: Determine what to do with measure proxy metric metadata")
+def test_metric_metadata_parsing_with_measure_proxy(multi_hop_join_semantic_model: SemanticModel) -> None:
+    """Tests internal metadata object parsing from a file into the Metric model object via measure proxy
+
+    The simple model has a broader array of metric definitions, but it does not appear to have a measure proxy
+    added via transformation. This test includes such a metric.
+
+    This only tests some basic file name parsing for each metric since they are not guaranteed
+    to be collected in the same file in the simple model, and the output here has been transformed
+    so the YAML contents might or might not match.
+    """
+    assert len(multi_hop_join_semantic_model.user_configured_model.metrics) > 0
+    for metric in multi_hop_join_semantic_model.user_configured_model.metrics:
+        assert (
+            metric.metadata is not None
+        ), f"Metadata should always be parsed out of the model, but None found for metric: {metric}!"
+        _assert_metadata_filename_is_valid(metric.metadata)
+
+
```

Comment:
At what point in the future do we decide on if we are doing this? Personally I'm on the side of "metrics created by proxy should get the metadata of the measure, and maybe something denoting it was created by proxy"

Reply:
We should decide this week and then we can open an issue to finalize the fixes. I don't want to push that across the MetricFlow/Transform boundary along with these changes since it affects parsing output, and we'll likely want to make Metadata required instead of optional in the model objects requesting it. But I also don't want this hanging in space forever, because if we really want Metadata required when set it's easier to do that now than later.